tests/colors/conversions.jl

```diff
@@ -1,30 +1,67 @@
 @testset "Colorspace transforms tests" begin
-    f = Chain(x->HSV.(x),channelify,flatten,Dense(768,16),Dense(16,10),x->σ.(x))
-    g_3 = Chain(Conv((3,3),7=>4,relu),
-              Conv((3,3),4=>3,relu),
-              x->colorify(RGB,x),
-              x->HSV.(x),
-              channelify,
-              flatten,
-              Dense(256*3,16),
-              Dense(16,10),
-              x->σ.(x))
-    g_1 = Chain(Conv((3,3),7=>4,relu),
-              Conv((3,3),4=>3,relu),
-              x->colorify(RGB,x),
-              x->Gray.(x),
-              channelify,
-              flatten,
-              Dense(256*1,16),
-              Dense(16,10),
-              x->σ.(x))
+    # f = Chain(x->HSV.(x),channelify,flatten,Dense(768,16),Dense(16,10),x->σ.(x))
+    # g_3 = Chain(Conv((3,3),7=>4,relu),
+    #           Conv((3,3),4=>3,relu),
+    #           x->colorify(RGB,x),
+    #           x->HSV.(x),
+    #           channelify,
+    #           flatten,
+    #           Dense(256*3,16),
+    #           Dense(16,10),
+    #           x->σ.(x))
+    # g_1 = Chain(Conv((3,3),7=>4,relu),
+    #           Conv((3,3),4=>3,relu),
+    #           x->colorify(RGB,x),
+    #           x->Gray.(x),
+    #           channelify,
+    #           flatten,
+    #           Dense(256*1,16),
+    #           Dense(16,10),
+    #           x->σ.(x))
+    ds = (7,7,5)
+    working_cspaces = (:YIQ,
+                       :LCHab,
+                       :Lab, 
+                       :BGRA,:ABGR,:BGR, 
+                       :RGBA,:ARGB,:RGB, 
+                       :HSL, 
+                       :AGray,:GrayA,:Gray, 
+                       :HSV) # colorspaces those have samplers defined for Base.Random
 
+    cspaces = (:HSV,:AHSV,:HSVA,
+               :Gray,:AGray,:GrayA,
+               :HSL,:AHSL,:HSLA,
+               :RGB,:ARGB,:RGBA,
+               :BGR,:ABGR,:BGRA,
+               :XYZ,:AXYZ,:XYZA,
+               :xyY,:AxyY,:xyYA,
+               :Lab,:ALab,:LabA,
+               :Luv,:ALuv,:LuvA,
+               :LCHab,:ALCHab,:LCHabA,
+               :LCHuv,:ALCHuv,:LCHuvA,
+               :DIN99,:ADIN99,:DIN99A,
+               :LMS,:ALMS,:LMSA,
+               :YIQ,:AYIQ,:YIQA) # tuple of all exprs of Colorspaces and their transparent variants
```

Comment:
You don't need to write them as `Symbol` and `rand(eval(cs), ds...)`, you can simply use plain types and use `rand(cs, ds...)`.

Reply:
Absolutely, I am just in the process of writing the `colorify` tests and then the differentiablilty tests. Goog thing you reminded me about reversibility, I had totally forgotten about it :)